### ISSUE TITLE
Нужно растягивать фон на 100%

### ISSUE THREAD

**Issue by frontgirl** [Sonrisa86/diplomaNetology#8]
https://github.com/Sonrisa86/diplomaNetology/blob/c59519a49964656995054f5cd0ecaf4406975353/css/main.css#L817

Не хватает background-size: cover.
Также сейчас абсолютно спозиционированный элемент позиционируется относительно всей страницы, а нужно, чтобы "родителем" был header. Для этого нужно задать блоку header position: relative. И обратите внимание, что :after должен быть не у .header, а у header - ведь именно этот блок у нас тянется на всю ширину.

**Comment by Sonrisa86** [comment]
спасибо!
я добавила  background-size: cover, и position, все получилось, :after у меня у header, да, я понимаю что он тянется.
если честно картинка выглядит не очень при варианте синий фон и полупрозрачная картинка.
если сравнить с футером, то картинка в футере намного четче получилась (это непрозрачная картинка и полупрозрачный синий фон).

**Comment by frontgirl** [comment]
Если вы посмотрите в макет, то обе картинки-маски там выполнены именно в таком виде (непрозрачный цвет  и полупрозрачная картинка поверх), так что принцип должен быть одинаковый и в хедере и в футере, к сожалению.

Почему я просила сделать :after именно на header, а не на .header: .header, как контейнер с текстом, сейчас находится "снизу" картинки(если вы прибавите opacity - увидите, что сейчас текст под ней) и чтобы это исправить, нужно с помощью z-index поставить маску "ниже" слоем(z-index: 0), чем текстовый контейнер. И в случае, если header:after - можно просто будет задать .header position: relative (чтобы работал z-index), тогда автоматически сработает значение z-index по умолчанию - 1 - и контейнер будет выше слоем, чем маска. С вашей текущей структурой либо нужно задавать position: relative каждому элементу в header, что плохо, потому что если я добавлю что-то туда - это уже не сработает, либо добавлять еще один контейнер только с этой целью.

Также хочу обратить ваше внимание, что на всех макетах обе маски (верхняя и нижняя) должны быть выполнены именно таким способом (возможно вы поняли, просто пока не начали делать, но я решила уточнить отдельно, потому что вижу на другой ширине экрана способ без :after).

**Comment by Sonrisa86** [comment]
картинки и принцип их расположения и в хедере и в футере я поменяю, без проблем, главное понять принцип.
Добавила обертку в хэдер (.header-wrapper), и в нее положила полупрозрачную картинку. Не срабатывает z-index. Ставлю position:relative, но для .header не срабатывает как ни кручу

**Comment by frontgirl** [comment]
Так, давайте разбираться. Я имела в виду z-index у элемента :before. Т.к. сейчас у вас маска сделана без :before, то z-index не нужен. Но это к сожалению, не самый оптимальный вариант, потому что у вас добавляется дополнительный div.

Давайте все-таки уберем третий элемент (оставляем header(можно ему добавить класс header) и div class="header-wrapper" - такое название мне очень нравится, т.к. отлично передает суть блока: это обертка для хедера.

У header class="header" делаем псевдоэлемент (можно :before, не обязательно делать :after).
Ему и зададим position: absolute и  z-index: 0; и не забудем растянуть его на 100% по ширине и высоте, как это у вас уже было сделано.
При этом блоку div class="header-wrapper" задаем position: relative.

**Comment by Sonrisa86** [comment]
Я видимо совсем тяжелый случай( 
все равно z-index не срабатывает.

**Comment by frontgirl** [comment]
Дело в том, что у вас сейчас стоит пробел между header и ":before", поэтому если вы посмотрите в инспекторе, где выводится :before - он вывелся у каждого дочернего элемента header: https://tppr.me/foT0r , целых 3 раза :( А нам нужен один.

Должна быть структура такая:

`<header class="header"><div class="header-wrapper"> ..... </div></header>`

и я предложила сделать :before у блока header:

`.header{
...
position: relative;
}
.header:before{
  content: "";
  position: absolute;
  z-index: 0;
 ..... /* остальные стили для позиционирования и ширины с высотой*/
}`

Видите, как маленькая опечатка может значительно повлиять на решение.

**Comment by Sonrisa86** [comment]
Вот же ж я дундук. 
Все исправила: и в футере и в хэдере все по-новому
